docs/spec/auth/vesting.md

```diff
@@ -0,0 +1,159 @@
+## Vesting
+
+### Intro and Requirements
+
+This paper specifies vesting account implementation for the Cosmos Hub. 
+The requirements for this vesting account is that it should be initialized during genesis with
+a starting balance X coins and a vesting endtime T. The owner of this account should be able to delegate to validators 
+and vote with locked coins, however they cannot send locked coins to other accounts until those coins have been unlocked. 
+The vesting account should also be able to spend any coins it receives from other users. 
+Thus, the bank module's `MsgSend` handler should error if a vesting account is trying to send an amount that exceeds their 
+unlocked coin amount.
+
+### Implementation
+
+##### Vesting Account implementation
+
+NOTE:  `Now = ctx.BlockHeader().Time`
+
+```go
+type VestingAccount interface {
+    Account
+    AssertIsVestingAccount() // existence implies that account is vesting.
+
+    // Calculates amount of coins that can be sent to other accounts given the current time
+    SendableCoins(sdk.Context) sdk.Coins
+}
+
+// Implements Vesting Account
+// Continuously vests by unlocking coins linearly with respect to time
+type ContinuousVestingAccount struct {
+    BaseAccount
+    OriginalCoins sdk.Coins // Coins in account on Initialization
+    ReceivedCoins sdk.Coins // Coins received from other accounts
+    SentCoins     sdk.Coins // Coins sent to other accounts
+
+    // StartTime and EndTime used to calculate how much of OriginalCoins is unlocked at any given point
+    StartTime     time.Time
+    EndTime       time.Time
+}
+
+// Uses time in context to calculate total unlocked coins
+SendableCoins(vacc ContinuousVestingAccount, ctx sdk.Context) sdk.Coins:
+    
+    // Coins unlocked by vesting schedule
+    unlockedCoins := ReceivedCoins - SentCoins + OriginalCoins * (Now - StartTime) / (EndTime - StartTime)
+
+    // Must still check for currentCoins constraint since some unlocked coins may have been delegated.
+    currentCoins := vacc.BaseAccount.GetCoins()
+
+    // min will return sdk.Coins with each denom having the minimum amount from unlockedCoins and currentCoins
+    return min(unlockedCoins, currentCoins)
+
+```
+
+The `VestingAccount` interface is used to assert that an account is a vesting account like so:
+
+```go
+vacc, ok := acc.(VestingAccount); ok
+```
+
+as well as to calculate the SendableCoins at any given moment.
+
+The `ContinuousVestingAccount` struct implements the Vesting account interface. It uses `OriginalCoins`, `ReceivedCoins`, 
+`SentCoins`, `StartTime`, and `EndTime` to calculate how many coins are sendable at any given point. 
+Since the vesting restrictions need to be implemented on a per-module basis, the `ContinuousVestingAccount` implements 
+the `Account` interface exactly like `BaseAccount`. Thus, `ContinuousVestingAccount.GetCoins()` will return the total of 
+both locked coins and unlocked coins currently in the account.
```

Comment:
Maybe good to add "Delegated coins are deducted from Account.GetCoins(), but do not count against unlocked coins because they are still at stake and will be reinstated (partially if slashed) after waiting the full unbonding period."